game/graphics/gfx.cpp

```diff
@@ -53,38 +54,179 @@ namespace Gfx {
+            controller_buttons_json[name].get<int>();
+      } else {
+        lg::warn(
+            "Controller button override not found for {}. Using controller default value: {}", name,
+            g_settings.pad_mapping_info.controller_button_mapping[controller_index][(int)button]);
+      }
+
+      if (keyboard_buttons_json.find(name) != keyboard_buttons_json.end()) {
+        g_settings.pad_mapping_info.keyboard_button_mapping[controller_index][(int)button] =
+            keyboard_buttons_json[name].get<int>();
+      } else {
+        lg::warn(
+            "Keyboard button override not found for {}. Using keyboard default value: {}", name,
+            g_settings.pad_mapping_info.keyboard_button_mapping[controller_index][(int)button]);
+      }
+    }
+
+    auto& keyboard_analogs_json = peripheral["Keyboard+Mouse"]["Analog"];
+    for (const auto& [name, value] : analog_map) {
+      Pad::AnalogMappingInfo analog_mapping;
+      if (keyboard_analogs_json[name].contains("Axis Id") == true) {
+        analog_mapping.mode = Pad::AnalogMappingMode::AnalogInput;
+        analog_mapping.axis_id = keyboard_analogs_json[name]["Axis Id"].get<int>();
+        g_settings.pad_mapping_info.keyboard_analog_mapping[controller_index][(int)value] =
+            analog_mapping;
+        continue;
+      }
+
+      if (keyboard_analogs_json[name].contains("Positive Key") == true) {
+        analog_mapping.positive_key = keyboard_analogs_json[name]["Positive Key"].get<int>();
+      } else {
+        lg::warn("Keyboard analog override not found for {}. Using keyboard default value: {}",
+                 name,
+                 g_settings.pad_mapping_info.keyboard_analog_mapping[controller_index][(int)value]
+                     .positive_key);
+      }
+
+      if (keyboard_analogs_json[name].contains("Negative Key") == true) {
+        analog_mapping.negative_key = keyboard_analogs_json[name]["Negative Key"].get<int>();
+      } else {
+        lg::warn("Keyboard analog override not found for {}. Using keyboard default value: {}",
+                 name,
+                 g_settings.pad_mapping_info.keyboard_analog_mapping[controller_index][(int)value]
+                     .negative_key);
+      }
+      g_settings.pad_mapping_info.keyboard_analog_mapping[controller_index][(int)value] =
+          analog_mapping;
+    }
+    g_settings.pad_mapping_info.mouse_x_axis_sensitivities[controller_index] =
+        peripheral["X-Axis Mouse Sensitivity"].get<double>();
+    g_settings.pad_mapping_info.mouse_y_axis_sensitivities[controller_index] =
+        peripheral["Y-Axis Mouse Sensitivity"].get<double>();
+    controller_index++;
+  }
+}
+
+void LoadSettings() {
+  auto filename = (file_util::get_user_config_dir() / "controller" / "controller-settings.json");
+  if (fs::exists(filename)) {
+    LoadPeripheralSettings(filename);
+    lg::info("Loaded graphics configuration file.");
+    return;
+  }
```

Comment:
I think you're missing something to create the settings file on the first run if it doesn't exist.
I added an `else { SavePeripheralSettings(); }` here and it seemed to work.

Reply:
I originally had a default mapping settings file generated if it couldn't find a controller-settings.json file. Although, it seemed like mandude wasn't very fond of the idea at the time. https://github.com/open-goal/jak-project/pull/1702#discussion_r938782693. I'll re-add the logic to create a new setting file back into the code.

Reply:
Hm, once there's an in game menu, maybe we can revisit this, but I think for now this is good. Otherwise I have no idea what all the things in the JSON should be called.